mapillary/utils/filter.py

```diff
@@ -371,3 +373,91 @@ def compass_angle(data: list, angles: tuple = (0.0, 360.0)) -> list:
         for feature in data
         if angles[0] <= feature["properties"]["compass_angle"] <= angles[1]
     ]
+
+
+def is_looking_at(image_feature: Feature, look_at_feature: Feature) -> bool:
+    """Return whether the image_feature is looking at the look_at_feature
+
+    :param image_feature: The feature set of the image
+    :type image_feature: dict
+
+    :param look_at_feature: The feature that is being looked at
+    :type look_at_feature: dict
+
+    :return: Whether the diff is greater than 310, or less than 50
+    :rtype: bool
+    """
+
+    # Pano accessible via the `get_image_layer`
+    # in config/api/vector_tiles.py
+    if image_feature["properties"]["is_pano"]:
+        return True
+
+    # Compass angle accessible via the `get_image_layer`
+    # in config/api/vector_tiles.py
+    if image_feature["properties"]["compass_angle"] < 0:
+        return False
+
+    # Getting the difference between the two provided GeoJSONs and the compass angle
+    diff: int = (
+        abs(
+            bearing(start=image_feature, end=look_at_feature)
+            - image_feature["properties"]["compass_angle"]
+        )
+        % 360
+    )
+
+    # If diff > 310 OR diff < 50
+    return 310 < diff or diff < 50
+
+
+def by_look_at_feature(image: dict, look_at_feature: Feature) -> bool:
+    """Filter through the given image features and return only features with the look_at_feature
+
+    :param image: The feature dictionary
+    :type image: dict
+
+    :param look_at_feature: Feature description
+    :type look_at_feature: A WGS84 GIS feature, TurfPy
+
+    :return: Whether the given feature is looking at the `look_at_features`
+    :rtype: bool
+    """
+
+    # Converting the coordinates in coords
+    coords = [image["geometry"]["coordinates"][0], image["geometry"]["coordinates"][1]]
+
+    # Getting the feature using `Feature`, `Point` from TurfPy
+    image_feature = Feature(
+        geometry=Point(coords, {"compass_angle": image["properties"]["compass_angle"]})
+    )
+
+    image_feature["properties"] = image["properties"]
+
+    # Does the `image_feature` look at the `look_at_feature`?
+    return is_looking_at(image_feature, look_at_feature)
+
+
+def hits_by_look_at(data: list, at: dict) -> list:
+    """Whether the given data have any feature that look at the `at` coordinates
+
+    :param data: List of features with an Image entity
+    :type data: list
+
+    :param at: The lng and lat coordinates. An example would be,
+        >>> at = {
+        >>>     'lng': <longitudinal parameter>,
+        >>>     'lat': <latitude parameter>,
+        >>> }
+    :type at: dict
+
+    :return: Filtered results of features only looking at `at`
+    :rtype: list
+    """
+
+    # Converting the `at` into a Feature object from TurfPy
+    at_feature = Feature(geometry=Point((at["lng"], at["lat"])))
+
+    return list(
+        filter(lambda image: by_look_at_feature(image, at_feature), data["features"])
+    )
```

Comment:
Can you explain the functionality of these 2 filters as well?

Reply:
These two again are implementations of the below code snippet that Christopher shared in #14. Extracting from the snippet, they are,

```typescript
function filterHitsByLookAt<T extends {_source: MapImage}>(hits: T[], lookat: {lng: number, lat: number}): T[] {
    const lookatFeature: any = turf.point([lookat.lng, lookat.lat]);
    return hits.filter(
        (image): boolean => {
            const coords = [image._source.l.lon, image._source.l.lat];
            const imageFeature = turf.point(coords, {'ca': image._source.ca});
            return isLookingAt(imageFeature, lookatFeature);
        },
    );
}
```

Sadly, `Python` lacks the functionality to write [clear, maintainable, multi-line, and easy to write lambda (anonymous functions)](https://stackoverflow.com/questions/1233448/no-multiline-lambda-in-python-why-not) as is written for the `hits.filter` functionality above. Thus, I went with writing a separate function on its own called `by_look_at_feature`.

Out of all the 3 functions implemented in `filters.py`, only `hits_by_looking_at` is 'registered' in `function_mapping` , thus, it is the only function externally exposed to users.